core/src/main/java/org/jruby/javasupport/proxy/JavaProxyClass.java

```diff
@@ -549,16 +550,14 @@ public static JavaProxyClass setProxyClassReified(final Ruby runtime, final Ruby
      */
     private static final ThreadLocal<Object[]> lookup = new ThreadLocal<>();
 
-    public static Integer addStaticInitLookup(Object... objects) {
+    public static int addStaticInitLookup(Object... objects) {
         // TODO: is this a log or an exception?
-        if (objects != null) ensureStaticIntConsumed();
+        if (objects != null) ensureStaticInitConsumed();
         lookup.set(objects);
-        if (objects == null) return 0;
-        int val = objects.hashCode();
-        return val;
+        return Arrays.hashCode(objects); // 0 if null
     }
 
-    public static void ensureStaticIntConsumed() {
```

Comment:
Technically, this wasn't actually a typo, as were were dealing with a "static int" in the generated class that must have the corresponding lookup consumed, but fine to change

Reply:
oh okay, thanks - did not realize the Int was intended.